### PR TITLE
Add Network UPS Tools to support Uninterruptable Power Supplies

### PR DESCRIPTION
**Pull request by florian-h05**
Part 1 of #1712.
Part 2 is Waveshare UPS HAT support, which I cannot contribute (at least for now).

## Description

This PR adds support for setting up Network UPS Tools both in unattended and interactive mode.

nut itself can be setup by openHABian to either connect to a local USB UPS (the code is ready to support more drivers, but I don't have any good proposals because `usbhid-ups` supports most modern UPSes in my feeling) or to another instance of nut on the local network, e.g. your Synology NAS.

The setup is currently not utilising `upssched`, which would allow to have mail notifications etc., but I think I can provide a follow up PR for this capability. I just don't want to blow up this PR anymore.

Note: the upsd users that are created have secure passwords that are generated on the fly by OpenSSL, except the one user that Synology NAS expects. Since this user cannot control anything, just read UPS data, I don't see a security problem there.

Note: I used the standard nut configuration files from Raspbian 11 and modified them to our needs. I decided against removing some sections, because this would avoid advanced users from manually editing these files.

## Testing

I have tested both the interactive and the unattended setup several times with:

- my Eaton 3S 700 directly connected to my dev Pi via USB
- my Eaton UPS connected to my Synology NAS, therefore using `netclient`

Ping @mstormi. BTW, I have seen your FOSDEM presentation, very interesting :+1:

### CODE REVIEW

File: openhabian-setup.sh [hunk 1]
```diff
@@ -124,6 +124,7 @@ if [[ -n "$UNATTENDED" ]]; then
   jsscripting_npm_install "openhab_rules_tools"
   zram_setup
   exim_setup
+  nut_setup
   permissions_corrections
```
Comment (mstormi) [inline openhabian-setup.sh:128]: setup without prior install ?

**Review by mstormi — COMMENTED** [review]
LGTM except for my comment at the end.

Did not test but when you say you did that's sufficient.

File: openhabian-setup.sh (hunk 1 above)
Comment (florian-h05) [reply openhabian-setup.sh:128]: I chose the same pattern as for the exim setup: the `nut_setup` methods installs nut if not already installed.

File: functions/nut.bash [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  fi
+
+  local upsmasterpw="$(openssl rand -base64 12)"
+  local upsslavepw="$(openssl rand -base64 12)"
+  local upsmonitor="MONITOR ups@localhost 1 upsmaster ${upsmasterpw} master"
+
+  local whiptailTitle="Network UPS Tools"
+  local introText="Network UPS Tools (nut) adds an Uniterruptable Power Supply (UPS) to your openHABian system.\\nIn case of a power outage, it protects your system from data loss by ensuring a safe shutdown.\\n\\nWe will guide you through the setup."
+
+  if ! nut_is_installed; then nut_install "install"; fi
```
Comment (florian-h05) [inline functions/nut.bash:70]: @mstormi Here it will install nut if it is not installed. I have tested this as well, works 👍

**Comment by florian-h05** [comment]
@mstormi Can this get merged?

**Review by mstormi — APPROVED** [review]

**Comment by florian-h05** [comment]
Thanks!

Only thing missing now is docs …

**Comment by mstormi** [comment]
> Only thing missing now is docs …

will you be contributing them, too ?

**Pull request merged**